examples_tests/60.ClusteredRendering/main.cpp

```diff
@@ -1,4 +1,3831 @@
+
+			video::IGPUSpecializedShader* const shaders[2] = { vertShader.get(), fragShader.get() };
+			renderpassIndep = logicalDevice->createGPURenderpassIndependentPipeline(
+				nullptr,
+				std::move(pipelineLayout),
+				shaders, shaders + 2,
+				vertexInputParams,
+				blendParams,
+				primitiveAssemblyParams,
+				rasterizationParams);
+		}
+
+		// graphics pipeline
+		{
+			video::IGPUGraphicsPipeline::SCreationParams creationParams = {};
+			creationParams.renderpassIndependent = renderpassIndep;
+			creationParams.renderpass = renderpass;
+			creationParams.subpassIx = DEBUG_DRAW_PASS_INDEX;
+			debugAABBGraphicsPipeline = logicalDevice->createGPUGraphicsPipeline(nullptr, std::move(creationParams));
+		}
+
+		// create debug draw AABB buffers
+		for (uint32_t i = 0u; i < FRAMES_IN_FLIGHT; ++i)
+		{
+			const size_t neededSSBOSize = VOXEL_COUNT_PER_LEVEL * LOD_COUNT * sizeof(nbl_glsl_shapes_AABB_t);
+
+			video::IGPUBuffer::SCreationParams creationParams = {};
+			creationParams.usage = static_cast<video::IGPUBuffer::E_USAGE_FLAGS>(video::IGPUBuffer::EUF_TRANSFER_DST_BIT | video::IGPUBuffer::EUF_STORAGE_BUFFER_BIT);
+			debugClustersForLightGPU[i] = logicalDevice->createDeviceLocalGPUBufferOnDedMem(creationParams, neededSSBOSize);
+		}
+
+		// create and update descriptor sets
+		{
+			// Todo(achal): Is creating 5 DS the only solution for:
+			// 1. Having a GPU buffer per command buffer? or,
+			// 2. Somehow sharing a GPU buffer amongst 5 different command buffers without
+			// race conditions
+			// Is using a property pool a solution?
+			const uint32_t setCount = FRAMES_IN_FLIGHT;
+			auto dsPool = logicalDevice->createDescriptorPoolForDSLayouts(video::IDescriptorPool::ECF_NONE, &dsLayout.get(), &dsLayout.get() + 1ull, &setCount);
+
+			for (uint32_t i = 0u; i < setCount; ++i)
+			{
+				debugAABBDescriptorSets[i] = logicalDevice->createGPUDescriptorSet(dsPool.get(), core::smart_refctd_ptr(dsLayout));
+
+				video::IGPUDescriptorSet::SWriteDescriptorSet writes[2];
+
+				writes[0].dstSet = debugAABBDescriptorSets[i].get();
+				writes[0].binding = 0u;
+				writes[0].count = 1u;
+				writes[0].arrayElement = 0u;
+				writes[0].descriptorType = asset::EDT_STORAGE_BUFFER;
+				video::IGPUDescriptorSet::SDescriptorInfo infos[2];
+				{
+					infos[0].desc = debugClustersForLightGPU[i];
+					infos[0].buffer.offset = 0u;
+					infos[0].buffer.size = debugClustersForLightGPU[i]->getCachedCreationParams().declaredSize;
+				}
+				writes[0].info = &infos[0];
+
+				writes[1].dstSet = debugAABBDescriptorSets[i].get();
+				writes[1].binding = 1u;
+				writes[1].count = 1u;
+				writes[1].arrayElement = 0u;
+				writes[1].descriptorType = asset::EDT_UNIFORM_BUFFER;
+				{
+					infos[1].desc = cameraUbo;
+					infos[1].buffer.offset = 0u;
+					infos[1].buffer.size = cameraUbo->getCachedCreationParams().declaredSize;
+				}
+				writes[1].info = &infos[1];
+				logicalDevice->updateDescriptorSets(2u, writes, 0u, nullptr);
+			}
+		}
+	}
+
```

Comment:
should be initialized in the constructor of and kept by a `nbl::ext::ClusteredLighting::CClusteredLighting` class